modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceClientCredentialTestCase.java

```diff
@@ -170,4 +205,114 @@ public void testValidateAccessToken() throws Exception {
+        Assert.assertEquals(errorObject.getHTTPStatusCode(), HttpStatus.SC_UNAUTHORIZED,
+                "Invalid access token response doesn't contain bad request status code.");
+        Assert.assertEquals(errorObject.getCode(), "invalid_client",
+                "Invalid access token response doesn't contain required error message.");
+        Assert.assertEquals(errorObject.getDescription(),
+                "A valid OAuth client could not be found for client_id: invalidConsumerKey",
+                "Invalid access token response doesn't contain required error description.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send client credentials token request with invalid client secret.",
+            dependsOnMethods = "testGetTokenUsingCCGrantWithInvalidClientId")
+    public void testGetTokenUsingCCGrantWithInvalidClientCredentials() throws Exception {
+
+        AuthorizationGrant clientCredentialsGrant = new ClientCredentialsGrant();
+        ClientID clientID = new ClientID(consumerKey);
+        Secret clientSecret = new Secret("invalidConsumerSecret");
+        ClientAuthentication clientAuth = new ClientSecretBasic(clientID, clientSecret);
+        Scope scope = new Scope(OAUTH2_SCOPE_OPENID, "xyz", VALID_RANDOM_SCOPE, INTERNAL_BULK_RESOURCE_CREATE);
+
+        URI tokenEndpoint = new URI(getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT, tenantInfo.getDomain()));
+        TokenRequest request = new TokenRequest(tokenEndpoint, clientAuth, clientCredentialsGrant, scope);
+        HTTPResponse tokenHTTPResp = request.toHTTPRequest().send();
+        Assert.assertNotNull(tokenHTTPResp, "Access token http response is null.");
+
+        TokenResponse tokenResponse = TokenResponse.parse(tokenHTTPResp);
+        Assert.assertFalse(tokenResponse.indicatesSuccess(),
+                "Token response indicated success. Token request should fail with invalid client id.");
+
+        TokenErrorResponse accessTokenResponse = (TokenErrorResponse) tokenResponse;
+        ErrorObject errorObject = accessTokenResponse.getErrorObject();
+        Assert.assertEquals(errorObject.getHTTPStatusCode(), HttpStatus.SC_UNAUTHORIZED,
+                "Invalid access token response doesn't contain bad request status code.");
+        Assert.assertEquals(errorObject.getCode(), "invalid_client",
+                "Invalid access token response doesn't contain required error message.");
+        Assert.assertEquals(errorObject.getDescription(),
+                "Client credentials are invalid.",
+                "Invalid access token response doesn't contain required error description.");
+    }
+
+    @Test(groups = "wso2.is", description = "Send client credentials token request without client secret.",
+            dependsOnMethods = "testGetTokenUsingCCGrantWithInvalidClientCredentials")
+    public void testGetTokenUsingCCGrantWithoutCredentials() throws Exception {
+
+
```

Comment:
Remove the additional line.

Reply:
fixed with https://github.com/wso2/product-is/pull/21449/commits/282d1fdb1dce22ea051a343452d7d8f3436b059f